### PR TITLE
(Card | OverlappingCard) Media container with multiple background colors

### PR DESCRIPTION
**Pull request by liorMar**
added multiple background colors to Media container

https://zeroheight.com/7sjjzhgo2/p/19342a/b/247036

### CODE REVIEW

**Comment by jonathanadler** [comment]
@liorMar 
Can you elaborate on this a bit more?
I'm not sure I understand this change.

**Comment by johannaasseraf** [comment]
Each container needs it’s own background color. According to this:
https://zeroheight.com/7sjjzhgo2/p/19342a/b/247036


On 10 June 2019 at 12:07:17, Jonathan Adler (notifications@github.com)
wrote:

@liorMar <https://github.com/liorMar>
Can you elaborate on this a bit more?
I'm not sure I understand this change.

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/wix/wix-ui-tpa/pull/99?email_source=notifications&email_token=AI3UXWB2GLCOVADNK23ZYRLPZYKUBA5CNFSM4HVVP4V2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXJK5TI#issuecomment-500346573>,
or mute the thread

File: src/components/Card/Card.st.css [hunk 1]
```diff
@@ -131,7 +132,7 @@
 
 .root > .infoContainer {
     box-sizing: border-box;
-    background-color: value(infoBGColor);
+    background-color: transparent;
```
Comment (jonathanadler) [inline src/components/Card/Card.st.css]: Why was this moved to the `root`?

File: src/components/OverlappingCard/OverlappingCard.st.css [hunk 2]
```diff
@@ -152,3 +146,9 @@
             BorderWidth value(borderWidth)
     );
 }
+
+.root > .mediaContainer {
+    background: linear-gradient(value(mediaBGColor), value(mediaBGColor)), linear-gradient(value(infoBGColor), value(infoBGColor));
```
Comment (jonathanadler) [inline src/components/OverlappingCard/OverlappingCard.st.css]: Where did the `linear-gradient`'s come from?

File: src/components/Card/Card.st.css (hunk 1 above)
Comment (liorMar) [reply src/components/Card/Card.st.css]: @jonathanadler for example if the user  wants to have transparent media background then we need to show the info background and not whats behind the Card so the easiest solution was to move the info background to root and make the info transparent

File: src/components/OverlappingCard/OverlappingCard.st.css (hunk 2 above)
Comment (liorMar) [reply src/components/OverlappingCard/OverlappingCard.st.css]: @jonathanadler in order to have the same behaviour as in `Card` we can't use the same solution. because of the height different. so `linear-gradient` lets us use multiple layers of background Color

**Comment by liorMar** [comment]
> @liorMar
> Can you elaborate on this a bit more?
> I'm not sure I understand this change.

the main goal is to have media container base Color Background the same as of Info container

File: src/components/Card/Card.st.css (hunk 1 above)
Comment (jonathanadler) [reply src/components/Card/Card.st.css]: So the info background color is the base background for the whole component?

**Comment by johannaasseraf** [comment]
There’s a background color for the whole card, and a background color for
the info container but bookings will set their info container background to
0% opacity


On 10 June 2019 at 18:10:04, Jonathan Adler (notifications@github.com)
wrote:

*@jonathanadler* commented on this pull request.
------------------------------

In src/components/Card/Card.st.css
<https://github.com/wix/wix-ui-tpa/pull/99#discussion_r292053266>:

> @@ -131,7 +132,7 @@

 .root > .infoContainer {
     box-sizing: border-box;
-    background-color: value(infoBGColor);
+    background-color: transparent;

So the info background color is the base background for the whole component?

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/wix/wix-ui-tpa/pull/99?email_source=notifications&email_token=AI3UXWGEAJRYO5H2ZUNKW3TPZZVERA5CNFSM4HVVP4V2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB3BSH6Q#discussion_r292053266>,
or mute the thread

File: src/components/Card/Card.st.css (hunk 1 above)
Comment (liorMar) [reply src/components/Card/Card.st.css]: > So the info background color is the base background for the whole component?

yes, but after Johanna's comment i think that i'll add a third css override for the Card BG with default the same as Info

File: src/components/OverlappingCard/OverlappingCard.st.css (hunk 2 above)
Comment (jonathanadler) [reply src/components/OverlappingCard/OverlappingCard.st.css]: I'm not familiar with this syntax.
Why are there 2 `linear-gradients`?
And what's the difference between this and `background-color: value(mediaBGColor);`

Comment (liorMar) [reply src/components/OverlappingCard/OverlappingCard.st.css]: you can't apply multiple layers of `background-color`  or have `background-color: value(mediaBGColor); ` and `background: linear-gradient(value(mediaBGColor), value(mediaBGColor));` at the same time.
`background` is stronger and will be shown on top of `bacgrounf-color` regardless the css order

Comment (liorMar) [reply src/components/OverlappingCard/OverlappingCard.st.css]: but `linear-gradient` after `linear-gradient` (only work with `background`) can let you apply multiple layers with order of how is on top.

Comment (jonathanadler) [reply src/components/OverlappingCard/OverlappingCard.st.css]: Why not use:
```
.media {
    background-color: fallback(value(mediaBgColor), value(cardBgColor));
}

.info {
    background-color: fallback(value(infoBgColor), value(cardBgColor));
}
```

Comment (liorMar) [reply src/components/OverlappingCard/OverlappingCard.st.css]: Because if the user selecting a Color with opacity the the `cardBgColor` will not be shown

Comment (jonathanadler) [reply src/components/OverlappingCard/OverlappingCard.st.css]: Thank you 🙏 
That was what I was missing.
